### PR TITLE
fix: adds labels to broker deployment [HYB-560]

### PR DESCRIPTION
**Pull request by saumilmac**
What:
- Adds additional labels to snyk-broker-deployment

### CODE REVIEW

**Review by soniqua — CHANGES_REQUESTED** [review]
Please add tests to validate this new functionality

File: charts/snyk-broker/templates/_helpers.tpl [hunk 1]
```diff
@@ -40,6 +40,9 @@ helm.sh/chart: {{ include "snyk-broker.chart" . }}
 app.kubernetes.io/version: {{ .Chart.AppVersion | quote }}
 {{- end }}
 app.kubernetes.io/managed-by: {{ .Release.Service }}
+{{- with .Values.labels }}
+{{  toYaml . }}
```
Comment (soniqua) [inline charts/snyk-broker/templates/_helpers.tpl:44]: q: How does this handle duplicate labels? May need to use a `merge` with override/discard any common labels that are already set.

**Review by soniqua — CHANGES_REQUESTED** [review]

File: charts/snyk-broker/templates/_helpers.tpl (hunk 1 above)
Comment (saumilmac) [reply charts/snyk-broker/templates/_helpers.tpl]: fixed

**Review by aarlaud — DISMISSED** [review]

File: charts/snyk-broker/templates/_helpers.tpl [hunk 2]
```diff
@@ -34,15 +34,16 @@ Create chart name and version as used by the chart label.
...
+{{- $commonLabels = merge $commonLabels (include "snyk-broker.selectorLabels" . | fromYaml) }}
 {{- if .Chart.AppVersion }}
-app.kubernetes.io/version: {{ .Chart.AppVersion | quote }}
+{{- $commonLabels = merge $commonLabels (dict "app.kubernetes.io/version" (quote .Chart.AppVersion)) }}
 {{- end }}
-app.kubernetes.io/managed-by: {{ .Release.Service }}
+{{- $commonLabels = merge $commonLabels (dict "app.kubernetes.io/managed-by" .Release.Service) }}
 {{- with .Values.labels }}
-{{  toYaml . }}
+{{- $commonLabels = merge $commonLabels . }}
```
Comment (soniqua) [inline charts/snyk-broker/templates/_helpers.tpl:44]: q: confirming this behaviour is as expected:

- Any labels provided in `.Values.labels` are overridden if they already exist in our standard definition
- Any new labels in `.Values.labels` are added

Comment (saumilmac) [reply charts/snyk-broker/templates/_helpers.tpl:44]: Correct, I merged the tests to add new labels and test if any labels provided in .Values.labels are overridden if they already exist. Snapshots showing that results.

File: charts/snyk-broker/values.yaml [hunk 3]
```diff
@@ -24,6 +24,9 @@ brokerDispatcherUrl: "https://api.snyk.io"
 # This number if only used if enableHighAvailabilityMode is true
 replicaCount: 2
 
+# Adds additional labels to broker deployment
```
Comment (soniqua) [inline charts/snyk-broker/values.yaml:27]: q: Does this impact the CRA deployment too?

File: charts/snyk-broker/tests/fixtures/default_values.yaml [hunk 4]
```diff
@@ -19,5 +19,4 @@ preflightChecks:
 
 highAvailabilityMode:
   enabled: false
-brokerDispatcherUrl: "https://api.test.snyk.io"
-
+brokerDispatcherUrl: "https://api.test.snyk.io"
```
Comment (soniqua) [inline charts/snyk-broker/tests/fixtures/default_values.yaml:22]: nit: Ensure there's a newline here

File: charts/snyk-broker/tests/broker_deployment_test.yaml [hunk 5]
```diff
@@ -33,3 +33,12 @@ tests:
...
       - matchSnapshot: {}
+  - it: handles duplicative labels and adds new labels
+    values:
+      - ./fixtures/default_values.yaml
+    set:
+      labels:
+        app.kubernetes.io/name: "my duplicated label"
+        test: node1
+    asserts:
+      - matchSnapshot: {}
```
Comment (soniqua) [inline charts/snyk-broker/tests/broker_deployment_test.yaml:44]: enh: asserting the label is present as opposed to an entire snapshot helps ensure this test is valid and not impacted by any future changes

File: charts/snyk-broker/values.yaml (hunk 3 above)
Comment (saumilmac) [reply charts/snyk-broker/values.yaml:27]: This doesn't impact CRA deployment. For CRA labels are static and not driven through the helper function 
https://github.com/snyk/snyk-broker-helm/blob/main/charts/snyk-broker/templates/cra_deployment.yaml#L7-L9

File: charts/snyk-broker/tests/fixtures/default_values.yaml (hunk 4 above)
Comment (saumilmac) [reply charts/snyk-broker/tests/fixtures/default_values.yaml]: updated

File: charts/snyk-broker/tests/broker_deployment_test.yaml (hunk 5 above)
Comment (saumilmac) [reply charts/snyk-broker/tests/broker_deployment_test.yaml]: I used `equals`, since labels are a map and `contains` require an array.

**Review by soniqua — APPROVED** [review]

**Pull request merged**